pulsar/internal/channel_cond.go

```diff
@@ -0,0 +1,73 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package internal
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"unsafe"
+)
+
+type ChCond struct {
+	L sync.Locker
+	n unsafe.Pointer
```

Comment:
A question about the name. What does `n` represent? It's better to give a meaningful name. See `sync.Cond`:

```go
type Cond struct {
        noCopy noCopy

        // L is held while observing or changing the condition
        L Locker

        notify  notifyList
        checker copyChecker
}
```

Reply:
I see `n` is a channel in other places like `Broadcast` and `WaitWithContext`. In this case, `n` is a short name of `notifyChan`. It's okay because it's recommended to have short variable names, see https://github.com/golang/go/wiki/CodeReviewComments#variable-names

But the rule is not applied for struct fields, because

>  the further from its declaration that a name is used, the more descriptive the name must be

The field could be used in the method definitions, so it's far from where it's declared (in the struct)

Reply:
`notifyChPtr` or `notifyChanPtr` would be better. ([`be short rather than long`](https://github.com/golang/go/wiki/CodeReviewComments#variable-names))